### PR TITLE
When using pg, create a schema when schemaTable specifies one

### PR DESCRIPTION
**Pull request by theasp**
When using pg, create a schema when schemaTable specifies one.  `schemaversion` will work just as it does now, but `postgrator.schemaversion` will place the table in the `postgrator` schema, creating it if it doesn't exist.

I've updated the tests to use `postgrator.schemaversion`.

### CODE REVIEW

**Comment by IvanJov** [comment]
Hey @theasp, thanks for the PR! I have couple questions:
1. Not sure why do you need this? Trying to understand what are the reasons to add a new schema
2. Can you use `const` or `let` instead of `var`?
3. Can you revert test changes and add new ones for this code? Since you added `schemaTable` in tests, tests will never check the non `schemaTable` code.

We will be connecting some CI to this repo in next days, we'll probably need to wait for that before merging this PR.

Thanks!

**Comment by theasp** [comment]
No problem @IvanJov

1. By default, when postgres creates a table it will put it in the `public` schema, unless there is a schema with the same name as your current role, in which case it will create the table there. This allows multiple users to share a database but have their own tables that don't conflict.  The `search_path` option controls which schemas are searched when you reference an object without a schema, and the schemas to list objects from:
```
mydb=# SHOW search_path;
   search_path   
-----------------
 "$user", public
```
Before switching to postgrator (I was using sqitch), I created a schema in my test database using the same name as the user, and postgrest created the table in the application user's schema.  In this case, if I was to use a different user to run postgrest, it wouldn't have access to the existing "schemaversion" table as it would be in the wrong schema for the user to access it, even though the existing table search would succeed.  When deploying to a fresh database though, postgrest will create the table in `public`.

I am working on an application that provides a search engine for an arbitrary database, I would like to use this feature to ensure that the `schemaversion` table doesn't appear in the `public` schema.  Following the convention described in [postgrest's](https://postgrest.com) documentation on [schema isolation](https://postgrest.com/en/v4.4/auth.html#schema-isolation), having a table in the `public` schema will control if the application can search the table or not.

Sqitch also creates a schema when using postgres:
https://github.com/theory/sqitch/blob/8457221d635baa20e07a13e4ee0255ecca5ace37/lib/App/Sqitch/Engine/pg.sql#L4

`"registry"` gets replaced with the name of the schema:
https://github.com/theory/sqitch/blob/8457221d635baa20e07a13e4ee0255ecca5ace37/lib/App/Sqitch/Engine/pg.pm#L213

tl;dr; not specifying the schema can cause the `schemaversion` table to wind up in an unexpected schema, and in my case, not the one I want.

2. Done!

3. Done!  It now looks like this:
```
  Driver: pg
    ✓ Migrates multiple versions up (000 -> 002) (920ms)
    ✓ Handles current version
    ✓ Has migration details in schema table
    ✓ Migrates one version up (002 -> 003 (55ms)
    ✓ Migrates generated SQL (67ms)
    ✓ Checksums generated SQL (50ms)
    ✓ Migrates to "max"
    ✓ Migrates down to 000 (120ms)
    ✓ Migrates to latest without input (661ms)
    ✓ Migrates down to 000 again (206ms)
    ✓ Errors on invalid md5 check (122ms)
    ✓ Migrates down to 000 again (69ms)
    ✓ Skips checksum validation if turned off (153ms)
    ✓ Migrates down to 000 again (122ms)

  Driver: pg (with schema)
    ✓ Migrates multiple versions up (000 -> 002) (185ms)
    ✓ Handles current version
    ✓ Has migration details in schema table
    ✓ Migrates one version up (002 -> 003 (48ms)
    ✓ Migrates generated SQL (66ms)
    ✓ Checksums generated SQL (51ms)
    ✓ Migrates to "max"
    ✓ Migrates down to 000 (239ms)
    ✓ Migrates to latest without input (136ms)
    ✓ Migrates down to 000 again (122ms)
    ✓ Errors on invalid md5 check (114ms)
    ✓ Migrates down to 000 again (244ms)
    ✓ Skips checksum validation if turned off (170ms)
    ✓ Migrates down to 000 again (130ms)
```

File: test/drivers/driverIntegration.js [hunk 1]
```diff
@@ -2,8 +2,8 @@
 const assert = require('assert')
 const Postgrator = require('../../postgrator')
 
-module.exports = function testConfig(config) {
-  describe(`Driver: ${config.driver}`, function() {
+module.exports = function testConfig(config, label) {
+  describe(label ? label : `Driver: ${config.driver}`, function() {
```
Comment (IvanJov) [inline test/drivers/driverIntegration.js]: Even shorter
```js
label || `Driver: ${config.driver}`
```

File: test/drivers/driverIntegration.js [hunk 2]
```diff
@@ -22,10 +22,11 @@ module.exports = function testConfig(config) {
     })
 
     it('Has migration details in schema table', function() {
+      var schemaTable = config.schemaTable || 'schemaversion'
```
Comment (IvanJov) [inline test/drivers/driverIntegration.js]: Please use `const`

**Comment by theasp** [comment]
Small issues should be dealt with.

**Comment by rickbergfalk** [comment]
Thanks @theasp for the PR and circleci config, and for reviewing this @IvanJov!

CircleCi is set up, and looks like it is doing its thing checking PRs. I gather not this one though since it is missing the config?

Out of curiosity, of that's pulled into this branch will CircleCi detect and build for this PR?

**Comment by IvanJov** [comment]
@theasp Can you update this PR with the latest master code, circleci should automatically detect config file and check the build. If it's green, @rickbergfalk can probably just merge it. Thank you!

**Comment by theasp** [comment]
@rickbergfalk I've pushed a squashed commit on top of the current master.  In CircleCI's "Advanced Settings" for the project, there is the option "Build forked pull requests", which may not be turned on.

**Comment by rickbergfalk** [comment]
Sorry forgot this was still out there - The build forked pull requests is turned on, but I'm guessing it wasn't detected since it was added after this was opened. 

I ran tests locally. Will publish shortly!

**Comment by theasp** [comment]
Awesome, thank you!
